File: src/main/node/com/example/Some.ts
/** Holds a present value. */
export default class Some<T> implements Option<T> {
    private readonly value: any;

    Some(value: any): void {
        // TODO
    }

    @Override
    isSome(): boolean {
        return /* TODO */;
    }

    @Override
    get(): any {
        return /* TODO */;
    }
}

File: src/main/java/com/example/Some.java
package com.example;

/** Holds a present value. */
public final class Some<T> implements Option<T> {
    private final T value;

    Some(T value) {
        this.value = value;
    }

    @Override
    public boolean isSome() {
        return true;
    }

    @Override
    public T get() {
        return value;
    }
}
